src/lib/holocene/orderable-list.svelte

```diff
@@ -14,26 +12,32 @@
     locked?: boolean;
   }
 
-  interface $$Props extends HTMLOlAttributes {
+  interface $$Props {
     items: OrderableItem[];
-    readonly?: boolean;
-    removeItem?: (index: number) => void;
-    addItem?: (index: number) => void;
+    availableItems?: OrderableItem[];
   }
 
-  let className: string = undefined;
-  export { className as class };
   export let items: OrderableItem[];
-  export let readonly: boolean = false;
-  export let removeItem: (index: number) => void = noop;
-  export let addItem: (index: number) => void = noop;
+  export let availableItems: OrderableItem[] = [];
+
+  const addItem = (index: number) => {
+    let tempAvailableItems = [...availableItems];
+    items = [...items, ...tempAvailableItems.splice(index, 1)];
+    availableItems = tempAvailableItems;
```

Comment:
Just a tiny nit, from a readability perspective it's hard to grok that items = [...items, ...tempAvailableItems.splice(index, 1)] is mutating tempAvailableItems as well for the next line assignment. Maybe (??) better like:


```
    let tempAvailableItems = [...availableItems].splice(index, 1);
    items = [...items, ...tempAvailableItems];
    availableItems = tempAvailableItems;

Reply:
Hm actually this may not work, because `.splice()` mutates the array but also returns the item(s) removed, so `tempAvailableItems` would just be an array of the 1 item that was spliced. I'll see if there's a more readable way to do this, we basically are just moving an item from one array to the other.

Reply:
How about

```typescript
  const addItem = (index: number) => {
    let tempAvailableItems = [...availableItems];
    const [removedItem] = tempAvailableItems.splice(index, 1);
    items = [...items, removedItem];
    availableItems = tempAvailableItems;
  };
```

Reply:
👍